### ISSUE TITLE
Unable to resolve module promise/setimmediate/es6-extensions

### ISSUE THREAD

**Issue by Harjot1Singh** [getsentry/sentry-react-native#2147]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?
@sentry/react-native@3.3.3 (iOS on react-native@0.67, Hermes enabled)

### Steps to Reproduce

1. Install the react-native SDK
2. Attempt to import it, and receive an error about being unable to resolve module promises
3. Clear all modules & pods, reset metro cache to be sure, and retry

### Actual Result

```
error: Error: Unable to resolve module promise/setimmediate/es6-extensions from /Volumes/Dev/Users/harjot/projects/ShabadOS-mobile/node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlers.js: promise/setimmediate/es6-extensions could not be found within the project or in these directories:
  node_modules

If you are sure the module exists, try these steps:
 1. Clear watchman watches: watchman watch-del-all
 2. Delete node_modules and run yarn install
 3. Reset Metro's cache: yarn start --reset-cache
 4. Remove the cache: rm -rf /tmp/metro-*
  46 |         const { polyfillGlobal, } = require("react-native/Libraries/Utilities/PolyfillFunctions");
  47 |         // Below, we follow the exact way React Native initializes its promise library, and we globally replace it.
> 48 |         const Promise = require("promise/setimmediate/es6-extensions");
     |                                  ^
  49 |         // As of RN 0.67 only done and finally are used
  50 |         require("promise/setimmediate/done");
  51 |         require("promise/setimmediate/finally");
    at ModuleResolver.resolveDependency (/Volumes/Dev/Users/harjot/projects/ShabadOS-mobile/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:129:15)
    at DependencyGraph.resolveDependency (/Volumes/Dev/Users/harjot/projects/ShabadOS-mobile/node_modules/metro/src/node-haste/DependencyGraph.js:288:43)
    at Object.resolve (/Volumes/Dev/Users/harjot/projects/ShabadOS-mobile/node_modules/metro/src/lib/transformHelpers.js:129:24)
    at resolve (/Volumes/Dev/Users/harjot/projects/ShabadOS-mobile/node_modules/metro/src/DeltaBundler/traverseDependencies.js:396:33)
    at /Volumes/Dev/Users/harjot/projects/ShabadOS-mobile/node_modules/metro/src/DeltaBundler/traverseDependencies.js:412:26
    at Array.reduce (<anonymous>)
    at resolveDependencies (/Volumes/Dev/Users/harjot/projects/ShabadOS-mobile/node_modules/metro/src/DeltaBundler/traverseDependencies.js:411:33)
    at processModule (/Volumes/Dev/Users/harjot/projects/ShabadOS-mobile/node_modules/metro/src/DeltaBundler/traverseDependencies.js:140:31)
    at async addDependency (/Volumes/Dev/Users/harjot/projects/ShabadOS-mobile/node_modules/metro/src/DeltaBundler/traverseDependencies.js:230:18)
    at async Promise.all (index 3)
```

**Comment by jennmueng** [comment]
@Harjot1Singh You should be able to fix this in the meantime by manually installing [`promise`](https://www.npmjs.com/package/promise). However, can I see your lockfile (`yarn.lock` or `package-lock.json`)? React Native should already have promise as a dependency.

**Comment by Harjot1Singh** [comment]
@jennmueng thanks for the response! `react-native` does have `promise@8.0.3` specified as a dependency, as seen here: https://github.com/shabados/mobile/blob/291b05784741a472c6291ebf54e1494f5b742d14/package-lock.json.

I'll manually install `promise` and see what happens.

**Comment by marandaneto** [comment]
@Harjot1Singh let us know if that worked.

**Comment by Harjot1Singh** [comment]
It did - might be to do with 8.0.3 vs 8.1.0?

**Comment by jennmueng** [comment]
> It did - might be to do with 8.0.3 vs 8.1.0?

Shouldn't be, the `setimmediate/es6-extensions` file has been there for quite a long time.

**Comment by marandaneto** [comment]
Closing it since it's apparently an environmental issue.
I can't reproduce on a brand new app and installing `promise` fixed the issue anyway.

**Comment by daay22** [comment]
When upgrading to Expo 53, I encountered this same issue. I've been stuck for weeks looking for an answer. Thank you for this post

**Comment by jg210** [comment]
The symptoms are a little different now:

```
Unable to resolve "promise/setimmediate/done" from "node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlersutils.js"
```
...but the bug is [the same](https://github.com/getsentry/sentry-react-native/issues/2147#issuecomment-1185549959) - the sentry RN code is using `promise` without declaring it in its package json.

**Comment by steinalex** [comment]
Seeing this problem today after upgrading React Native to `0.79.5` (as part of an Expo 53 upgrade). 

Manually adding `"promise": "8.3.0"` to my dependencies resolved it, but as mentioned above, I think it should added as a dependency of `@sentry/react-native`

**Comment by rossparachute** [comment]
> Seeing this problem today after upgrading React Native to `0.79.5` (as part of an Expo 53 upgrade).
> 
> Manually adding `"promise": "8.3.0"` to my dependencies resolved it, but as mentioned above, I think it should added as a dependency of `@sentry/react-native`

Ditto, had this happen on @sentry/react-native:7.0.1 and had to manually install promise:8.3.0.

**Comment by DTreg1** [comment]
Hitting this on `@sentry/react-native@7.2.0` + `react-native@0.81.5` + Expo SDK 54 + iOS 26.5 sim (2026-05-27). Error:

\`\`\`
Unable to resolve "promise/setimmediate/done" from "node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlersutils.js"
\`\`\`

`promise` is installed under `node_modules/react-native/node_modules/promise/` (as a transitive of RN), but not hoisted to the top-level, so Metro's resolution from inside `@sentry/react-native` (which walks `node_modules/@sentry/react-native/node_modules` → `node_modules`) can't find it.

Workaround that fixes it: add `"promise": "^8.3.0"` as a direct dependency in the consuming app's `package.json`, forcing it to hoist.

Confirms reports above from @rossparachute (7.0.1) and @steinalex (RN 0.79.5) — the bug recurs with each transitive-dep tree rearrangement and the underlying gap (`@sentry/react-native` uses `promise` without declaring it) is still present.

Filing this comment rather than a fresh issue since the canonical thread is here, but happy to open a dedicated tracking issue if that'd help triage.

**Comment by kevinlinv** [comment]
Opened another issue to hopefully revisit this here: https://github.com/getsentry/sentry-react-native/issues/6614
